gio/src/unix_fd_list.rs

```diff
@@ -47,39 +47,43 @@ pub trait UnixFDListExtManual: sealed::Sealed + IsA<UnixFDList> + Sized {
     }
 
     #[doc(alias = "g_unix_fd_list_get")]
-    fn get(&self, index_: i32) -> Result<RawFd, glib::Error> {
+    fn get(&self, index_: i32) -> Result<OwnedFd, glib::Error> {
         unsafe {
             let mut error = ptr::null_mut();
-            let ret = ffi::g_unix_fd_list_get(self.as_ref().to_glib_none().0, index_, &mut error);
+            let raw_fd =
+                ffi::g_unix_fd_list_get(self.as_ref().to_glib_none().0, index_, &mut error);
+            let fd = OwnedFd::from_raw_fd(raw_fd);
             if error.is_null() {
-                Ok(ret)
+                Ok(fd)
             } else {
                 Err(from_glib_full(error))
             }
         }
     }
 
     #[doc(alias = "g_unix_fd_list_peek_fds")]
-
-    fn peek_fds(&self) -> Vec<RawFd> {
+    fn peek_fds(&self) -> Vec<BorrowedFd<'_>> {
```

Comment:
This can actually just return `&[BorrowedFd]`. You can use `std::slice::from_raw_parts()` plus a cast. `BorrowedFd` is guaranteed to have the same layout as an actual fd.